internal/services/eventhub/eventhub_resource.go

```diff
@@ -66,9 +68,39 @@ func resourceEventHub() *pluginsdk.Resource {
 			},
 
 			"message_retention": {
-				Type:         pluginsdk.TypeInt,
-				Required:     true,
-				ValidateFunc: validate.ValidateEventHubMessageRetentionCount,
+				Type:          pluginsdk.TypeInt,
+				Optional:      true,
+				ValidateFunc:  validate.ValidateEventHubMessageRetentionCount,
+				ConflictsWith: []string{"retention_description"},
+			},
+
+			"retention_description": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*schema.Schema{
+						"cleanup_policy": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(eventhubs.CleanupPolicyRetentionDescriptionDelete),
+								string(eventhubs.CleanupPolicyRetentionDescriptionCompact),
+							}, false),
+						},
+
+						"retention_time_in_hours": {
+							Type:     pluginsdk.TypeInt,
+							Optional: true,
+						},
+
+						"tombstone_retention_time_in_hours": {
+							Type:     pluginsdk.TypeInt,
+							Optional: true,
+						},
```

Comment:
is it possible for both these to be set at the same time? If not, can we add some validation around that here?